### ISSUE TITLE
File Save As does not work in JupyerLab 2.1.0

### ISSUE THREAD

**Issue by dclong** [jupyterlab/jupyterlab#8216]
## Description

The "Save As..." menu fails to work sometimes. 

## Reproduce

1. Have JupyterLab 2.1.0 installed. You can try the Docker image [dclong/jupyterlab](https://github.com/dclong/docker-jupyterlab) with the tag "200408" for convenience. 

```
docker run -it \
    --hostname jupyterlab \
    --log-opt max-size=50m \
    -p 8888:8888 \
    -e DOCKER_USER=$(id -un) \
    -e DOCKER_USER_ID=$(id -u) \
    -e DOCKER_PASSWORD=$(id -un) \
    -e DOCKER_GROUP_ID=$(id -g) \
    -v $(pwd):/workdir \
    -v $(dirname $HOME):/home_host \
    dclong/jupyterlab:200408 /scripts/sys/init.sh
```
1. Open a text file (markdown or a Python script).
2. Click on File -> Save (Markdown) As...
3. Type in a new name for the file.
4. You will see that the new file does not appear in the navigation panel. 

## Expected behavior

I expect a new file to be generated when using the "Save As" menu.

## Context

- Operating System and version: Docker with Ubuntu 19.04 
- Browser and version: Chrome 80.0
- JupyterLab version: 2.1.0

<details><summary>Troubleshoot Output</summary>
<pre>
```
$ jupyter troubleshoot
$PATH:
        /usr/local/sbin
        /usr/local/bin
        /usr/sbin
        /usr/bin
        /sbin
        /bin
        /usr/games
        /usr/local/games

sys.path:
        /usr/local/bin
        /usr/lib/python37.zip
        /usr/lib/python3.7
        /usr/lib/python3.7/lib-dynload
        /usr/local/lib/python3.7/dist-packages
        /usr/lib/python3/dist-packages

sys.executable:
        /usr/bin/python3

sys.version:
        3.7.3 (default, Oct  7 2019, 12:56:13) 
        [GCC 8.3.0]

platform.platform():
        Linux-4.19.76-linuxkit-x86_64-with-Ubuntu-19.04-disco

which -a jupyter:
        /usr/local/bin/jupyter

pip list:
        Package                Version        
        ---------------------- ---------------
        astroid                2.3.3          
        attrs                  19.3.0         
        autopep8               1.5.1          
        backcall               0.1.0          
        bleach                 3.1.4          
        certifi                2020.4.5.1     
        chardet                3.0.4          
        colorama               0.4.3          
        decorator              4.4.2          
        defusedxml             0.6.0          
        entrypoints            0.3            
        flake8                 3.7.9          
        gitdb                  4.0.2          
        GitPython              3.1.0          
        idna                   2.9            
        importlib-metadata     1.6.0          
        ipykernel              5.2.0          
        ipython                7.13.0         
        ipython-genutils       0.2.0          
        ipywidgets             7.5.1          
        isort                  4.3.21         
        jedi                   0.15.2         
        Jinja2                 2.11.1         
        json5                  0.9.4          
        jsonschema             3.2.0          
        jupyter                1.0.0          
        jupyter-client         6.1.2          
        jupyter-console        6.1.0          
        jupyter-core           4.6.3          
        jupyterlab             2.1.0          
        jupyterlab-server      1.1.0          
        lazy-object-proxy      1.4.3          
        MarkupSafe             1.1.1          
        mccabe                 0.6.1          
        mistune                0.8.4          
        more-itertools         8.2.0          
        mypy                   0.770          
        mypy-extensions        0.4.3          
        nbconvert              5.6.1          
        nbdime                 2.0.0          
        nbformat               5.0.5          
        notebook               6.0.3          
        packaging              20.3           
        pandocfilters          1.4.2          
        parso                  0.6.2          
        pexpect                4.8.0          
        pickleshare            0.7.5          
        pip                    18.1           
        pluggy                 0.13.1         
        prometheus-client      0.7.1          
        prompt-toolkit         3.0.5          
        ptyprocess             0.6.0          
        py                     1.8.1          
        pycodestyle            2.5.0          
        pydocstyle             5.0.2          
        pyflakes               2.1.1          
        Pygments               2.6.1          
        PyGObject              3.32.0         
        pylint                 2.4.4          
        pyparsing              2.4.7          
        pyrsistent             0.16.0         
        pytest                 5.4.1          
        python-apt             1.8.5-ubuntu0.3
        python-dateutil        2.8.1          
        python-jsonrpc-server  0.3.4          
        python-language-server 0.31.9         
        pyzmq                  19.0.0         
        qtconsole              4.7.2          
        QtPy                   1.9.0          
        requests               2.23.0         
        rope                   0.16.0         
        Send2Trash             1.5.0          
        setuptools             40.8.0         
        six                    1.14.0         
        smmap                  3.0.1          
        snowballstemmer        2.0.0          
        terminado              0.8.3          
        testpath               0.4.4          
        tornado                6.0.4          
        traitlets              4.3.3          
        typed-ast              1.4.1          
        typing-extensions      3.7.4.2        
        ujson                  1.35           
        urllib3                1.25.8         
        wcwidth                0.1.9          
        webencodings           0.5.1          
        wheel                  0.32.3         
        widgetsnbextension     3.5.1          
        wrapt                  1.11.2         
        xinstall               0.6.2          
        yapf                   0.29.0         
        zipp                   3.1.0
```
</pre>
</details>

<details><summary>Command Line Output</summary>
<pre>
```
docker run -it \
>     --hostname jupyterlab \
>     --log-opt max-size=50m \
>     -p 8888:8888 \
>     -e DOCKER_USER=$(id -un) \
>     -e DOCKER_USER_ID=$(id -u) \
>     -e DOCKER_PASSWORD=$(id -un) \
>     -e DOCKER_GROUP_ID=$(id -g) \
>     -v $(pwd):/workdir \
>     -v $(dirname $HOME):/home_host \
>     dclong/jupyterlab:next /scripts/sys/init.sh
id: ‘chdu’: no such user
Adding user chdu to group sudo
[I 11:03:16.662 LabApp] Writing notebook server cookie secret to /home/chdu/.local/share/jupyter/runtime/notebook_cookie_secret
[I 11:03:17.899 LabApp] JupyterLab extension loaded from /usr/local/lib/python3.7/dist-packages/jupyterlab
[I 11:03:17.899 LabApp] JupyterLab application directory is /usr/local/share/jupyter/lab
[I 11:03:18.030 LabApp] Serving notebooks from local directory: /workdir
[I 11:03:18.030 LabApp] The Jupyter Notebook is running at:
[I 11:03:18.031 LabApp] http://jupyterlab:8888/?token=fa8c5e7e8765564ab76adffa4691083790fa187cea317612
[I 11:03:18.031 LabApp]  or http://127.0.0.1:8888/?token=fa8c5e7e8765564ab76adffa4691083790fa187cea317612
[I 11:03:18.031 LabApp] Use Control-C to stop this server and shut down all kernels (twice to skip confirmation).
[C 11:03:18.039 LabApp] 
    
    To access the notebook, open this file in a browser:
        file:///home/chdu/.local/share/jupyter/runtime/nbserver-40-open.html
    Or copy and paste one of these URLs:
        http://jupyterlab:8888/?token=fa8c5e7e8765564ab76adffa4691083790fa187cea317612
     or http://127.0.0.1:8888/?token=fa8c5e7e8765564ab76adffa4691083790fa187cea317612
[W 11:03:26.518 LabApp] Clearing invalid/expired login cookie username-localhost-8888
[W 11:03:26.519 LabApp] Clearing invalid/expired login cookie username-localhost-8888
[W 11:03:26.520 LabApp] Clearing invalid/expired login cookie username-localhost-8888
[W 11:03:26.522 LabApp] Clearing invalid/expired login cookie username-localhost-8888
[W 11:03:26.522 LabApp] Forbidden
[W 11:03:26.523 LabApp] 403 GET /api/kernels?1586801006489 (172.17.0.1) 6.05ms referer=http://localhost:8888/lab?
[W 11:03:26.524 LabApp] Forbidden
[W 11:03:26.525 LabApp] 403 GET /api/terminals?1586801006491 (172.17.0.1) 5.63ms referer=http://localhost:8888/lab?
[W 11:03:26.526 LabApp] Forbidden
[W 11:03:26.527 LabApp] 403 GET /api/sessions?1586801006490 (172.17.0.1) 6.91ms referer=http://localhost:8888/lab?
[W 11:03:26.528 LabApp] Forbidden
[W 11:03:26.529 LabApp] 403 GET /api/contents/?content=1&1586801006492 (172.17.0.1) 8.21ms referer=http://localhost:8888/lab?
[I 11:03:26.783 LabApp] 302 GET /lab? (172.17.0.1) 0.87ms
[I 11:05:03.046 LabApp] 302 GET /?token=fa8c5e7e8765564ab76adffa4691083790fa187cea317612 (172.17.0.1) 2.60ms
[I 11:05:04.973 LabApp] Build is up to date
[W 11:05:24.724 LabApp] 404 GET /api/contents/nima.md?1586801124703 (172.17.0.1): No such file or directory: nima.md
[W 11:05:24.724 LabApp] No such file or directory: nima.md
[W 11:05:24.724 LabApp] 404 GET /api/contents/nima.md?1586801124703 (172.17.0.1) 2.51ms referer=http://127.0.0.1:8888/lab
[I 11:06:19.413 LabApp] Writing notebook-signing key to /home/chdu/.local/share/jupyter/notebook_secret
[W 11:06:19.415 LabApp] Notebook Untitled.ipynb is not trusted
[I 11:06:19.905 LabApp] Kernel started: 177e953b-2406-405e-951c-54fa1f382786
[I 11:06:20.566 LabApp] Starting buffering for 177e953b-2406-405e-951c-54fa1f382786:10a3d947-1ff4-4bd6-a86b-697aa1e003d8
[W 11:06:27.844 LabApp] 404 GET /api/contents/Untitled2.ipynb?1586801187819 (172.17.0.1): No such file or directory: Untitled2.ipynb
[W 11:06:27.844 LabApp] No such file or directory: Untitled2.ipynb
[W 11:06:27.845 LabApp] 404 GET /api/contents/Untitled2.ipynb?1586801187819 (172.17.0.1) 2.51ms referer=http://127.0.0.1:8888/lab
[W 11:06:27.902 LabApp] 404 GET /api/contents/Untitled2.ipynb?content=0&1586801187874 (172.17.0.1): No such file or directory: Untitled2.ipynb
[W 11:06:27.903 LabApp] No such file or directory: Untitled2.ipynb
[W 11:06:27.903 LabApp] 404 GET /api/contents/Untitled2.ipynb?content=0&1586801187874 (172.17.0.1) 2.80ms referer=http://127.0.0.1:8888/lab
[I 11:06:27.915 LabApp] Uploading file to /Untitled2.ipynb
[W 11:06:27.915 LabApp] Notebook Untitled2.ipynb is not trusted
[I 11:06:35.593 LabApp] Starting buffering for 177e953b-2406-405e-951c-54fa1f382786:c821d0ff-f136-4046-88f3-7e5a08ce634f
[W 11:06:47.061 LabApp] 404 GET /api/contents/notes2.md?1586801207048 (172.17.0.1): No such file or directory: notes2.md
[W 11:06:47.061 LabApp] No such file or directory: notes2.md
[W 11:06:47.062 LabApp] 404 GET /api/contents/notes2.md?1586801207048 (172.17.0.1) 2.91ms referer=http://127.0.0.1:8888/lab
[I 11:18:11.189 LabApp] New terminal with automatic name: 1
TermSocket.open: 1
TermSocket.open: Opened 1

```
</pre>
</details>

<details><summary>Browser Output</summary>
<pre>
</pre>
</details>

**Comment by krassowski** [comment]
I can confirm this issue.
- "Save notebook as" works ok,
- Renaming txt files works ok,
- "Saving as" does NOT work for plain txt files

Nothing in browser console apart from failed contents request:

`http://localhost:8888/api/contents/some_path/new_name.txt?1586994000490 404 (Not Found)`

payload:

`{"message": "No such file or directory: some_path/new_name.txt", "reason": null}`

I do not know how "save as" is implemented, but I do not see a request I could associate with it.

**Comment by quangIO** [comment]
`this.sessionContext.session` is null in this case. My assumption is that files without any kernel association will not get `session`(Connection) initialized.
https://github.com/jupyterlab/jupyterlab/blob/f0153e0258b32674c9aec106383ddf7b618cebab/packages/docregistry/src/context.ts#L780-L792

**Comment by jasongrout** [comment]
> `this.sessionContext.session` is null in this case. My assumption is that files without any kernel association will not get `session`(Connection) initialized.

Thanks for narrowing this down. A fix is at #8238
